server/src/main/java/org/opensearch/search/startree/filter/provider/DimensionFilterMapper.java

```diff
@@ -0,0 +1,413 @@
+import org.opensearch.search.startree.filter.RangeMatchDimFilter;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.opensearch.index.mapper.NumberFieldMapper.NumberType.BYTE;
+import static org.opensearch.index.mapper.NumberFieldMapper.NumberType.DOUBLE;
+import static org.opensearch.index.mapper.NumberFieldMapper.NumberType.FLOAT;
+import static org.opensearch.index.mapper.NumberFieldMapper.NumberType.HALF_FLOAT;
+import static org.opensearch.index.mapper.NumberFieldMapper.NumberType.INTEGER;
+import static org.opensearch.index.mapper.NumberFieldMapper.NumberType.LONG;
+import static org.opensearch.index.mapper.NumberFieldMapper.NumberType.SHORT;
+import static org.opensearch.index.mapper.NumberFieldMapper.NumberType.hasDecimalPart;
+import static org.opensearch.index.mapper.NumberFieldMapper.NumberType.signum;
+
+/**
+ * Generates the @{@link DimensionFilter} raw values and the @{@link MappedFieldType} of the dimension.
+ */
+@ExperimentalApi
+public interface DimensionFilterMapper {
+    /**
+     * Generates @{@link ExactMatchDimFilter} from Term/Terms query input.
+     * @param mappedFieldType:
+     * @param rawValues:
+     * @return :
+     */
+    DimensionFilter getExactMatchFilter(MappedFieldType mappedFieldType, List<Object> rawValues);
+
+    /**
+     * Generates @{@link RangeMatchDimFilter} from Range query input.
+     * @param mappedFieldType:
+     * @param rawLow:
+     * @param rawHigh:
+     * @param includeLow:
+     * @param includeHigh:
+     * @return :
+     */
+    DimensionFilter getRangeMatchFilter(
+        MappedFieldType mappedFieldType,
+        Object rawLow,
+        Object rawHigh,
+        boolean includeLow,
+        boolean includeHigh
+    );
+
+    /**
+     * Called during conversion from parsedUserInput to segmentOrdinal for every segment.
+     * @param dimensionName:
+     * @param value:
+     * @param starTreeValues:
+     * @param matchType:
+     * @return :
+     */
+    Optional<Long> getMatchingOrdinal(
+        String dimensionName,
+        Object value,
+        StarTreeValues starTreeValues,
+        DimensionFilter.MatchType matchType
+    );
+
+    /**
+     * Singleton Factory for @{@link DimensionFilterMapper}
+     */
+    class Factory {
+
+        private static final Map<NumberFieldMapper.NumberType, DimensionFilterMapper> NUMERIC_SINGLETON_MAPPINGS = Map.of(
+            BYTE,
+            new IntegerFieldMapperNumeric(),
+            SHORT,
+            new IntegerFieldMapperNumeric(),
+            INTEGER,
+            new IntegerFieldMapperNumeric(),
+            LONG,
+            new SignedLongFieldMapperNumeric(),
+            HALF_FLOAT,
+            new HalfFloatFieldMapperNumeric(),
+            FLOAT,
+            new FloatFieldMapperNumeric(),
+            DOUBLE,
+            new DoubleFieldMapperNumeric()
+        );
```

Comment:
Somewhat similar to the comment for the query-based filters, I think it makes sense to unify this under a `Map<String, DimensionFilterMapper>`, then rely on the (unique) type names.

That way, in future, we'll be able to easily register named types from plugins to their own `DimensionFilterMapper`s.